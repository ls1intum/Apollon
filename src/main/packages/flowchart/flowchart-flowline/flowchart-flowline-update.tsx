import React, { Component, ComponentClass } from 'react';
import { connect } from 'react-redux';
import { compose } from 'redux';
import styled from 'styled-components';
import { Button } from '../../../components/controls/button/button';
import { Divider } from '../../../components/controls/divider/divider';
import { ExchangeIcon } from '../../../components/controls/icon/exchange';
import { TrashIcon } from '../../../components/controls/icon/trash';
import { Textfield } from '../../../components/controls/textfield/textfield';
import { Header } from '../../../components/controls/typography/typography';
import { I18nContext } from '../../../components/i18n/i18n-context';
import { localized } from '../../../components/i18n/localized';
import { ModelState } from '../../../components/store/model-state';
import { UMLElementRepository } from '../../../services/uml-element/uml-element-repository';
import { UMLRelationshipRepository } from '../../../services/uml-relationship/uml-relationship-repository';
import { FlowchartFlowline } from './flowchart-flowline';

const Flex = styled.div`
  display: flex;
  align-items: baseline;
  justify-content: space-between;
`;

class FlowchartFlowlineUpdateComponent extends Component<Props> {
  render() {
    const { element } = this.props;

    return (
      <div>
        <section>
          <Flex>
            <Header gutter={false} style={{ flexGrow: 1 }}>
              {this.props.translate('packages.Flowchart.FlowchartFlowline')}
            </Header>
            <Button color="link" onClick={() => this.props.flip(element.id)}>
              <ExchangeIcon />
            </Button>
            <Button color="link" onClick={() => this.props.delete(element.id)}>
              <TrashIcon />
            </Button>
          </Flex>
          <Divider />
        </section>
        <section>
          <Textfield value={element.name} onChange={this.rename} autoFocus />
        </section>
      </div>
    );
  }

  private rename = (name: string) => {
    this.props.update(this.props.element.id, { name });
  };
}

type OwnProps = {
  element: FlowchartFlowline;
};

type StateProps = {};

type DispatchProps = {
  update: typeof UMLElementRepository.update;
  delete: typeof UMLElementRepository.delete;
  flip: typeof UMLRelationshipRepository.flip;
};

type Props = OwnProps & StateProps & DispatchProps & I18nContext;

const enhance = compose<ComponentClass<OwnProps>>(
  localized,
  connect<StateProps, DispatchProps, OwnProps, ModelState>(null, {
    update: UMLElementRepository.update,
    delete: UMLElementRepository.delete,
    flip: UMLRelationshipRepository.flip,
  }),
);

export const FlowchartFlowlineUpdate = enhance(FlowchartFlowlineUpdateComponent);
