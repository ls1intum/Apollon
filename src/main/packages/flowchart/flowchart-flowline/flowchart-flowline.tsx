import { FlowchartRelationshipType } from '..';
import { UMLRelationship } from '../../../services/uml-relationship/uml-relationship';

export class FlowchartFlowline extends UMLRelationship {
  type = FlowchartRelationshipType.FlowchartFlowline;
}
