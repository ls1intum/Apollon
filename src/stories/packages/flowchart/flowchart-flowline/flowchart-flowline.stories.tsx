import { Meta, Story } from '@storybook/react/types-6-0';
import React from 'react';
import { FlowchartFlowline } from '../../../../main/packages/flowchart/flowchart-flowline/flowchart-flowline';
import {
  Props,
  FlowchartFlowlineComponent,
} from '../../../../main/packages/flowchart/flowchart-flowline/flowchart-flowline-component';

export default {
  title: 'Packages/Flowchart/Flowchart Flowline',
  component: FlowchartFlowlineComponent,
  args: { isSvg: true },
} as Meta;

const Template: Story<Props> = (args) => <FlowchartFlowlineComponent {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  element: new FlowchartFlowline({
    bounds: { x: 0, y: 0, width: 200, height: 100 },
    id: '1',
    name: 'Flowchart Flowline',
    owner: null,
  }),
};
