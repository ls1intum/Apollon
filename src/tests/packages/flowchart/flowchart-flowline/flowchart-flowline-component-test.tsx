import * as React from 'react';
import { render } from '@testing-library/react';
import { FlowchartFlowline } from '../../../../main/packages/flowchart/flowchart-flowline/flowchart-flowline';
import { FlowchartFlowlineComponent } from '../../../../main/packages/flowchart/flowchart-flowline/flowchart-flowline-component';

it('render the flowchart-flowline-component', () => {
  const flowline: FlowchartFlowline = new FlowchartFlowline({
    id: 'd37b8ce3-17d2-4432-8fff-6c38ff2a1334',
    name: 'TestFlowchartComponent',
  });
  const { getByText, baseElement } = render(
    <svg>
      <FlowchartFlowlineComponent element={flowline} />
    </svg>,
  );
  expect(getByText(flowline.name)).toBeInTheDocument();
  expect(baseElement).toMatchSnapshot();
});
