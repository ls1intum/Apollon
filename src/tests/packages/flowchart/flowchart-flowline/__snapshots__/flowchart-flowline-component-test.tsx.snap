// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the flowchart-flowline-component 1`] = `
<body>
  <div>
    <svg>
      <g>
        <marker
          id="marker-d37b8ce3-17d2-4432-8fff-6c38ff2a1334"
          markerHeight="30"
          markerUnits="strokeWidth"
          markerWidth="22"
          orient="auto"
          refX="30"
          refY="15"
          stroke-dasharray="1,0"
          viewBox="0 0 30 30"
        >
          <path
            d="M0,29 L30,15 L0,1"
            fill="none"
            stroke="black"
          />
        </marker>
        <polyline
          fill="none"
          marker-end="url(#marker-d37b8ce3-17d2-4432-8fff-6c38ff2a1334)"
          points="0 0,200 100"
          stroke="black"
          stroke-width="1"
        />
        <text
          dominant-baseline="text-after-edge"
          dy="-5"
          pointer-events="none"
          text-anchor="middle"
          x="100"
          y="50"
        >
          TestFlowchartComponent
        </text>
      </g>
    </svg>
  </div>
</body>
`;
